### PR TITLE
blog: distributed tracing in ceph using jaeger blog 2021

### PR DESCRIPTION
**Pull request by ideepika**
Signed-off-by: Deepika Upadhyay <dupadhya@redhat.com>

### CODE REVIEW

File: src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md
Comment (idryomov) [inline src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md:96]: For me, this Bluejeans recording has the screen split into two halves, with the meeting screen on the left and the actual demo on the right.  I would suggest something like asciinema instead -- much better suited for blog posts IMO.

File: src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md
Comment (idryomov) [inline src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md:108]: dev@ceph.io -- ceph-devel is for kernel client work.

File: src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md
Comment (idryomov) [inline src/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger.md:44]: ... using what?

**Review by idryomov — COMMENTED** [review]
Just a couple of things that caught my eye -- I didn't proof-read or anything like that.

**Comment by ideepika** [comment]
thankyou @idryomov! appreciate your review! will amend it.

**Comment by adamduncan** [comment]
Hi @ideepika 👋 

Just a quick heads up re: filenaming. We'll want to ensure the blog post is named `index.md`, as per the [Folders and filenaming](https://github.com/ceph/ceph.io/tree/main/src/en/news/blog#folders-and-file-naming) info that can be found in the Blog's README.

This will ensure that the blog appears at `/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/` (`index.html` being inferred default page) and not `/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger/`, as it would with file and folder named in this way.

Hope that helps

**Comment by ideepika** [comment]
> Hi @ideepika 
> 
> Just a quick heads up re: filenaming. We'll want to ensure the blog post is named `index.md`, as per the [Folders and filenaming](https://github.com/ceph/ceph.io/tree/main/src/en/news/blog#folders-and-file-naming) info that can be found in the Blog's README.
> 
> This will ensure that the blog appears at `/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/` (`index.html` being inferred default page) and not `/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/distributed-tracing-in-ceph-using-jaeger/`, as it would with file and folder named in this way.
> 
> Hope that helps

thanks @adamduncan, have renamed! maybe it must be good to add in github comment while opening a PR :)

**Comment by adamduncan** [comment]
No problem, @ideepika. Blog posts (and most other collected pages, e.g. events, etc.) will need to include frontmatter data to set the title, date, author, etc. for the post. Again, this is documented in the [Blog README](https://github.com/ceph/ceph.io/tree/main/src/en/news/blog#frontmatter).

Worth referencing the other posts that exist currently to get a feel for how they make use of this, and ensure that this post has the fields required prior to merging. Thanks.

**Comment by adamduncan** [comment]
FYI re: tags: https://github.com/ceph/ceph.io/issues/166 (looking for better ways to document this, so first-time contributors have a clear approach to work to). Appreciate you bearing with us as we get the new site and workflows off the ground 🙂

**Comment by ideepika** [comment]
@adamduncan is gif not supported? https://wip-jaeger-blog.ceph.io/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/ 
I can't see it rendered. Can you let me know why youtube link might be erroring `This content is blocked. Contact the site owner to fix the issue.`

**Comment by adamduncan** [comment]
>  is gif not supported?

Ah, good spot. You're correct. The site processes all images in the site, to ensure they're optimised. Currently that is handling jpg, png and svg, but not gif. Have opened a ticket (#243), and will try to get that in for you ASAP.

> Can you let me know why youtube link might be erroring

Looks like the `iframe` embed being used currently has a `src` of the video shortlink (`https://youtu.be/GVXlBL6vtVY`). When embedding the video (by copying the embed snippet from YouTube), you'll get a different embed URL. E.g. `https://www.youtube.com/embed/GVXlBL6vtVY`.

To make embedding videos from YouTube easier however, we've created a snippet for use throughout the site (again, we've documented [YouTube embed code](https://github.com/ceph/ceph.io#youtube-video-player-embed) in the project README). This is the best way to embed videos from YouTube:

```
{% YouTube 'GVXlBL6vtVY', 'Quick Demo of using JaegerUI for Ceph' %}
```

**Comment by ideepika** [comment]
> Ah, good spot. You're correct. The site processes all images in the site, to ensure they're optimised. Currently, that is handling jpg, png and svg, but not gif. Have opened a ticket (#243), and will try to get that in for you ASAP.

thanks! :)
> {% YouTube 'GVXlBL6vtVY', 'Quick Demo of using JaegerUI for Ceph' %}

Awesome!

**Comment by adamduncan** [comment]
@ideepika #243 to add support for GIF assets has now been resolved with PR #244. If you'd like to rebase this branch from `main` to test against your blog post, that'd be great. 👍

**Comment by ideepika** [comment]
> @ideepika #243 to add support for GIF assets has now been resolved with PR #244. If you'd like to rebase this branch from `main` to test against your blog post, that'd be great.

@adamduncan thanks! I rebased and repushed but I cannot see gif rendered, am I missing something? https://wip-jaeger-blog.ceph.io/en/news/blog/2021/

**Review by zdover23 — APPROVED** [review]
I am going to approve this PR. After this PR has been merged, I will make another PR in which I will make a small number of stylistic and syntactic changes.

The "What's Next" section is the only section that requires a rewrite on account of syntactically inadmissible text. Everything else created here is clean enough to be merged proudly.

In the main, this is excellent developer-written documentation.

**Comment by ideepika** [comment]
thanks @zdover23 ! i want to make some changes in master that might be helpful, will merge once these nits are resolved

**Comment by adamduncan** [comment]
jenkins test

**Comment by adamduncan** [comment]
> I rebased and repushed but I cannot see gif rendered, am I missing something?

@ideepika I don't think so. I'm not sure the branch deploy has rebuilt yet, so the post at https://wip-jaeger-blog.ceph.io/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/ still hasn't inherited the GIF fix.

FYI — there's an update to the build config in the works https://github.com/ceph/ceph-build/pull/1869 that should ensure PR deploys rebuild, and we can then see the latest.

**Comment by adamduncan** [comment]
Ah, @ideepika, I hadn't realised you've created a new PR for this, based on a fork of the `main` branch.

We were still looking at links to your previous (now closed) PR `wip-jaeger-blog`. That's why we weren't seein the changes.

I've reviewed from https://main.ceph.io/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/ and can confirm that the blog post is rendering as expected, GIFs and all.

Would you like to confirm and merge this PR to make the blog post live when you're ready?

**Comment by ideepika** [comment]
> Ah, @ideepika, I hadn't realised you've created a new PR for this, based on a fork of the `main` branch.
> 
> We were still looking at links to your previous (now closed) PR `wip-jaeger-blog`. That's why we weren't seein the changes.
> 
> I've reviewed from https://main.ceph.io/en/news/blog/2021/distributed-tracing-in-ceph-using-jaeger/ and can confirm that the blog post is rendering as expected, GIFs and all.
> 
> Would you like to confirm and merge this PR to make the blog post live when you're ready?

@adamduncan  aah I am so used to named branches, sorry forgot I made a PR again main branch, force pushed wip-jaeger-blog as main

**Comment by liewegas** [comment]
Is this ready to post?

**Comment by ideepika** [comment]
@liewegas blog is good to go, but without https://github.com/ceph/ceph/pull/38783 build would fail if someone tries it out, the cmake pr is close to merge I think

**Comment by ideepika** [comment]
> I am going to approve this PR. After this PR has been merged, I will make another PR in which I will make a small number of stylistic and syntactic changes.
> 
> The "What's Next" section is the only section that requires a rewrite on account of syntactically inadmissible text. Everything else created here is clean enough to be merged proudly.
> 
> In the main, this is excellent developer-written documentation.

@zdover23 merged the PR, please feel free to update anything relevant. thanks!

**Pull request merged**
